From 25caf45221bad5ee6930c2278959695575540ea0 Mon Sep 17 00:00:00 2001
From: Kevin <kw22@princeton.edu>
Date: Thu, 14 Jan 2021 14:34:26 -0500
Subject: [PATCH] fixed statuscmd to account for status2d

---
 dwm/dwm.c | 45 +++++++++++++++++++++++++++++++++++++++++++--
 1 file changed, 43 insertions(+), 2 deletions(-)

diff --git a/dwm/dwm.c b/dwm/dwm.c
index f5e87ab..d7ecc63 100644
--- a/dwm/dwm.c
+++ b/dwm/dwm.c
@@ -225,6 +225,7 @@ static void showhide(Client *c);
 static void sigchld(int unused);
 static void sigdwmblocks(const Arg *arg);
 static void spawn(const Arg *arg);
+static int status2dtextlength(const char *stext);
 static void tag(const Arg *arg);
 static void tagmon(const Arg *arg);
 static void tile(Monitor *);
@@ -465,7 +466,7 @@ buttonpress(XEvent *e)
 			arg.ui = 1 << i;
 		} else if (ev->x < x + blw)
 			click = ClkLtSymbol;
-		else if (ev->x > (x = selmon->ww - (int)TEXTW(stext) + lrpad)) {
+		else if (ev->x > (x = selmon->ww - status2dtextlength(stext))) {
 			click = ClkStatusText;
 
 			char *text = rawstext;
@@ -476,7 +477,7 @@ buttonpress(XEvent *e)
 				if ((unsigned char)text[i] < ' ') {
 					ch = text[i];
 					text[i] = '\0';
-					x += TEXTW(text) - lrpad;
+					x += status2dtextlength(text);
 					text[i] = ch;
 					text += i + 1;
 					i = -1;
@@ -1947,6 +1948,46 @@ spawn(const Arg *arg)
 	}
 }
 
+int
+status2dtextlength(const char *stext)
+{
+	int i, w, len;
+	short isCode = 0;
+	char *text;
+	char *p;
+
+	len = strlen(stext) + 1;
+	if (!(text = (char *)malloc(sizeof(char)*len)))
+		die("malloc");
+	p = text;
+	copyvalidchars(text, (char *)stext);
+
+	/* compute width of the status text */
+	w = 0;
+	i = -1;
+	while (text[++i]) {
+		if (text[i] == '^') {
+			if (!isCode) {
+				isCode = 1;
+				text[i] = '\0';
+				w += TEXTW(text) - lrpad;
+				text[i] = '^';
+				if (text[++i] == 'f')
+					w += atoi(text + ++i);
+			} else {
+				isCode = 0;
+				text = text + i + 1;
+				i = -1;
+			}
+		}
+	}
+	if (!isCode)
+		w += TEXTW(text) - lrpad;
+
+	free(p);
+	return w;
+}
+
 void
 tag(const Arg *arg)
 {
-- 
2.30.0

